File: ms-java/src/main/java/com/jakubeeee/playground/msjava/organization/internal/DepartmentEntity.java
package com.jakubeeee.playground.msjava.organization.internal;

import com.jakubeeee.playground.common.jpa.IdentifiableEntity;
import org.jetbrains.annotations.NotNull;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

import static java.util.Objects.requireNonNull;

@Entity
@Table(name = "DEPARTMENTS")
public class DepartmentEntity extends IdentifiableEntity {

    @Column(name = "NAME", nullable = false)
    private String name;

    protected DepartmentEntity() {
        // no-arg constructor required by the JPA
    }

    protected DepartmentEntity(@NotNull String name) {
        this.name = requireNonNull(name);
    }

    public String name() {
        return name;
    }

}

File: ms-kotlin/src/main/kotlin/com/jakubeeee/playground/mskotlin/organization/internal/DepartmentEntity.kt
package com.jakubeeee.playground.mskotlin.organization.internal

import com.jakubeeee.playground.common.jpa.IdentifiableEntity
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Table

@Entity
@Table(name = "DEPARTMENTS")
open class DepartmentEntity protected constructor() : IdentifiableEntity() {

    @Column(name = "NAME", nullable = false)
    open var name: String? = null
        protected set

    constructor(name: String) : this() {
        this.name = name
    }

}